src/Worker.Extensions.DurableTask/DurableOrchestrator.cs

```diff
@@ -0,0 +1,114 @@
+    /// <exception cref="ArgumentNullException">Thrown if <paramref name="triggerStateJson"/> or <paramref name="orchestratorFunc"/> is <c>null</c>.</exception>
+    public static string LoadAndRun<TInput, TOutput>(string triggerStateJson, Func<TaskOrchestrationContext, TInput?, Task<TOutput?>> orchestratorFunc)
+    {
+        if (orchestratorFunc == null)
+        {
+            throw new ArgumentNullException(nameof(orchestratorFunc));
+        }
+
+        FuncTaskOrchestrator<TInput, TOutput> orchestrator = new(orchestratorFunc);
+        return LoadAndRun(triggerStateJson, orchestrator);
+    }
+
+    /// <summary>
+    /// Deserializes orchestration history from <paramref name="triggerStateJson"/> and uses it to resume the orchestrator
+    /// implemented by <paramref name="implementation"/>.
+    /// </summary>
+    /// <param name="triggerStateJson">The trigger state of the orchestrator function. This is expected to be a JSON string but should be treated as internal and opaque.</param>
+    /// <param name="implementation">An <see cref="ITaskOrchestrator"/> implementation that defines the orchestrator logic.</param>
+    /// <returns>Returns a serialized set of orchestrator actions that should be used as the return value of the orchestrator function trigger.</returns>
+    /// <exception cref="ArgumentNullException">Thrown if <paramref name="triggerStateJson"/> or <paramref name="implementation"/> is <c>null</c>.</exception>
+    /// <exception cref="ArgumentException">Thrown if <paramref name="triggerStateJson"/> contains invalid data.</exception>
+    public static string LoadAndRun(string triggerStateJson, ITaskOrchestrator implementation)
+    {
+        if (string.IsNullOrEmpty(triggerStateJson))
+        {
+            throw new ArgumentNullException(nameof(triggerStateJson));
+        }
+
+        if (implementation == null)
+        {
+            throw new ArgumentNullException(nameof(implementation));
+        }
+
+        OrchestratorState? state = JsonConvert.DeserializeObject<OrchestratorState>(triggerStateJson, SerializerSettings);
+        if (state?.PastEvents == null || state?.NewEvents == null)
+        {
+            throw new ArgumentException(
+                $"The {nameof(triggerStateJson)} payload contained invalid data. If this data came from the orchestration trigger binding, it indicates a mismatch between the binding extension used by this app and the WebJobs binding extension being used by the Functions host.",
+                nameof(triggerStateJson));
+        }
+
+        FunctionsWorkerContext workerContext = new(JsonDataConverter.Default);
+
+        // Re-construct the orchestration state from the history.
+        OrchestrationRuntimeState runtimeState = new(state.PastEvents);
+        foreach (HistoryEvent newEvent in state.NewEvents)
+        {
+            runtimeState.AddEvent(newEvent);
+        }
```

Comment:
Is there a reason why we shouldn't incorporate this for-loop inside the constructor of `OrchestrationRuntimeState`?